dask/backends.py

```diff
@@ -119,7 +120,15 @@ def decorator(
 
             @wraps(fn)
             def wrapper(*args, **kwargs):
-                return getattr(self, dispatch_name)(*args, **kwargs)
+                func = getattr(self, dispatch_name)
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    raise RuntimeError(
+                        f"An error occurred while calling the {funcname(func)} "
+                        f"method registered to the {backend} backend.\n"
+                        f"Original Message: {e}"
+                    ) from e
```

Comment:
If I unederstand correctly, this wont work for the cudf backend (and your new test should fail). Note that `backend` will always corrspond to "pandas" (even when the cudf backend is active), because that variable corresponds to the backend which registered the dispatchable function "in place" within `dask.dataframe` (which is always "pandas").

Reply:
Ah, I see -- you're totally right. This still feels like the right place to emit this error. Would the `_config_field` lookup be the thing we want?